src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1080,6 +1080,28 @@ protected override bool IsValidPath(string path)
                 }
             }
 
+            // .NET introduced a change where invalid characters are accepted https://learn.microsoft.com/en-us/dotnet/core/compatibility/2.1#path-apis-dont-throw-an-exception-for-invalid-characters
+            // We need to check for invalid characters ourselves
+
+            // validate the filename portion first
+            if (Path.GetFileName(path).IndexOfAny(Path.GetInvalidFileNameChars()) != -1)
+            {
+                return false;
+            }
+
+            // now validate each segment of the path
+            var directory = Path.GetDirectoryName(path);
+            if (directory is not null)
+            {
+                foreach (string segment in directory.Split(Path.DirectorySeparatorChar))
+                {
+                    if (segment.IndexOfAny(Path.GetInvalidPathChars()) != -1)
```

Comment:
Shouldn't this still be `GetInvalidFileNameChars`? This will not check for things like `*`, `"`, etc in all path segments. I'm not sure you need `GetInvalidPathChars()` at all and should just do a split on the original `path` and check each segment with `GetInvalidFileNameChars`.

Reply:
Looking at the dotnet code, `GetInvalidPathChars()` is _currently_ a strict subset of `GetInvalidFilenameChars()`, so it seems there's no value in calling `GetInvalidPathChars()` at all and using `GetInvalidFilenameChars()` for the segments makes sense.  I'll make that change.